### PR TITLE
add WithTaskListMapper and WithWorkflowTasklistMapper utils for workflow executions

### PR DESCRIPTION
**Pull request by shijiesheng**
**What changed?**

* Added util functions to change tasklists (activity tasklists and workflow tasklists)

**Why?**

This is needed for internal usage to support tenancy separation

**How did you test it?**

unit test

**Potential risks**

### CODE REVIEW

File: internal/workflow.go [hunk 1]
```diff
@@ -1413,6 +1413,13 @@ func WithWorkflowTaskList(ctx Context, name string) Context {
 	return ctx1
 }
 
+// WithWorkflowTaskListApply returns a copy of context and changes workflow tasklist with apply function.
```
Comment (taylanisikdemir) [inline internal/workflow.go:1416]: nit: maybe a better name is `WithWorkflowTaskListMapper`

File: internal/workflow.go [hunk 2]
```diff
@@ -1413,6 +1413,13 @@ func WithWorkflowTaskList(ctx Context, name string) Context {
...
+func WithWorkflowTaskListApply(ctx Context, apply func(*string) *string) Context {
```
Comment (taylanisikdemir) [inline internal/workflow.go:1417]: we don't need to pass the pointer to this callback func.
`func(string) string` would be more user friendly

File: internal/workflow.go (hunk 1 above)
Comment (taylanisikdemir) [reply internal/workflow.go:1416]: same in other places

**Review by taylanisikdemir — APPROVED** [review]

File: internal/workflow.go (hunk 1 above)
Comment (shijiesheng) [reply internal/workflow.go]: fixed

File: internal/workflow.go (hunk 2 above)
Comment (shijiesheng) [reply internal/workflow.go]: make sense

**Comment by Groxx** [comment]
er. what's the purpose here / how is it intended to be used?

**Pull request merged**